mongo/client.go

```diff
@@ -348,6 +351,12 @@ func (c *Client) endSessions(ctx context.Context) {
 }
 
 func (c *Client) configure(opts *options.ClientOptions) error {
+	var defaultOptions = 0
```

Comment:
Are we expecting more default options in the future?
```suggestion
	var defaultOptions int
```
Firstly, I think you can just use the zero-value to initialize here. But, are we expecting more default options in the future? Even if we were, I might have a hard time remembering to increment this value for new ones. I think you can just require `len(serverOpts) > 3` on L700 and note that `// server options: WithClock, WithConnectionOptions and WithMaxPoolSize` are included in the default server options.

Reply:
The original logic on [L700](https://github.com/mongodb/mongo-go-driver/blob/5c6942e0f0ee496dc0fcbca2d5a5cb597ada3154/mongo/client.go#L700) confuses me. It seems not even accept a customer `maxPoolSize` setting (excluding the default option logic) to me. If this is the case, a counter is more clear than twisted `if` conditions in my opinion even if it is the only default value.